junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TempDirectoryTests.java

```diff
@@ -766,4 +773,30 @@ void createReadonlyFile(@TempDir Path tempDir) throws IOException {
 
 	}
 
+	// https://github.com/junit-team/junit5/issues/2079
+	static class TempDirCanBeUsedInNestedTestClass {
+
+		@TempDir
+		File tempDir;
+
+		@Nested
+		class NestedTestClass {
+
+			@Test
+			void testNested() {
```

Comment:
I recommend adding the same test to the top-level class as well, just for the sake of completeness.

Reply:
Sorry for the late reply. I will apply your suggestions as soon as I'll find a little bit time.